app/assets/javascripts/discourse/app/components/post-list/item/index.gjs

```diff
@@ -23,44 +32,136 @@ export default class PostListItem extends Component {
     }
   }
 
+  get hiddenClass() {
+    return (
+      this.args.post.hidden && !(this.currentUser && this.currentUser.staff)
+    );
+  }
+
+  get deletedClass() {
+    return this.args.post.deleted ? "deleted" : "";
+  }
+
+  get user() {
+    return {
+      id: this.args.post.user_id,
+      name: this.args.post.name,
+      username: this.args.post.draft_username || this.args.post.username,
+      avatar_template: this.args.post.avatar_template,
+      title: this.args.post.user_title,
+      primary_group_name: this.args.post.primary_group_name,
+    };
+  }
+
+  get postId() {
+    return this.args.post.id || this.args.post.post_id;
+  }
+
+  @action
+  buildOneboxes(element) {
+    schedule("afterRender", () => {
+      loadOneboxes(
```

Comment:
I think this API was only really designed for use in the composer. Using it on a list of many posts could trigger a whole load of AJAX requests, and may lead to rate limiting. Plus, IIRC the onebox API is only available to authenticated users?

I would've expected us to be using `cooked` HTML here with the oneboxes already included. So it's surprising that we're having to add them in JS 🤔